sympy/sets/ordinals.py

```diff
@@ -0,0 +1,260 @@
+from sympy.core import Basic, Integer
+from sympy.core.compatibility import with_metaclass
+from sympy.core.singleton import Singleton, S
+import operator
+
+class OmegaPower(Basic):
+    """
+    Represents ordinal exponential and multiplication terms one of the
+    building blocks of the Ordinal class.
+    In OmegaPower(a, b) a represents exponent and b represents multiplicity.
+    """
+    def __new__(cls, a, b):
+        if isinstance(b, int):
+            b = Integer(b)
+        if not isinstance(b, Integer) or b <= 0:
+            raise TypeError("multiplicity must be a positive integer")
+
+        if not isinstance(a, Ordinal):
+            try:
+                a = Ordinal.convert(a)
+            except TypeError:
+                return NotImplemented
```

Comment:
It seems that `NotImplemented` is intended as a return value of [comparisons and binary operations](https://docs.python.org/2/library/stdtypes.html?highlight=notimplemented#the-notimplemented-object) only. I think that no `try` block is necessary. If `convert` raises a `TypeError`, then that could passed to the caller (as in case of argument `b`).